lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfNmdaBaseServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+import org.opendaylight.yangtools.yang.data.impl.schema.builder.impl.ImmutableAnydataNodeBuilder;
+import org.opendaylight.yangtools.yang.data.util.ImmutableNormalizedAnydata;
+import org.opendaylight.yangtools.yang.model.api.DataSchemaNode;
+import org.opendaylight.yangtools.yang.model.api.SchemaContext;
+
+public class NetconfNmdaBaseServiceImpl extends NetconfBaseServiceImpl implements NetconfNmdaBaseService {
+
+    public static final QName NETCONF_NMDA_EXTENSION_QNAME = QName.create("urn:ietf:params:xml:ns:yang:ietf-netconf-nmda", "2019-01-07", "ietf-netconf-nmda").intern();
+    public static final QName NETCONF_GET_DATA_QNAME = QName.create(NETCONF_NMDA_EXTENSION_QNAME, "get-data");
+    public static final QName NETCONF_EDIT_DATA_QNAME = QName.create(NETCONF_NMDA_EXTENSION_QNAME, "edit-data").intern();
+    private static final YangInstanceIdentifier.NodeIdentifier NETCONF_FILTER_NODEID =
+            YangInstanceIdentifier.NodeIdentifier.create(QName.create(NETCONF_GET_DATA_QNAME, "subtree-filter"));
+    private static final YangInstanceIdentifier.NodeIdentifier NETCONF_FILTER_CHOICE_NODEID =
+            YangInstanceIdentifier.NodeIdentifier.create(QName.create(NETCONF_GET_DATA_QNAME, "filter-spec"));
+    private static final YangInstanceIdentifier.NodeIdentifier NETCONF_ORIGIN_FILTERS_CHOICE_NODEID =
+            YangInstanceIdentifier.NodeIdentifier.create(QName.create(NETCONF_GET_DATA_QNAME, "origin-filters"));
+    private static final QName NETCONF_OPERATION_QNAME_LEGACY = NETCONF_OPERATION_QNAME.withoutRevision().intern();
+    private static final YangInstanceIdentifier.NodeIdentifier NETCONF_EDIT_DATA_CONFIG_NODEID =
+            YangInstanceIdentifier.NodeIdentifier.create(QName.create(NETCONF_EDIT_DATA_QNAME, "config"));
+
+    private final DOMRpcService domRpcService;
+    private final SchemaContext schemaContext;
+
+    public NetconfNmdaBaseServiceImpl(NodeId nodeId, DOMRpcService domRpcService, SchemaContext schemaContext) {
+        super(nodeId, domRpcService, schemaContext);
+        this.domRpcService = domRpcService;
+        this.schemaContext = schemaContext;
+    }
+
+    @Override
+    public ListenableFuture<? extends DOMRpcResult> getData(QName sourceDatastore,
+                                                            Optional<YangInstanceIdentifier> filterYII,
+                                                            Optional<Boolean> configFilter,
+                                                            Optional<Integer> maxDepth,
+                                                            Optional<Set<QName>> originFilter,
+                                                            Optional<Boolean> negateOriginFilter,
+                                                            Optional<Boolean> withOrigin) {
+        List<DataContainerChild<?, ?>> getDataChildren = new ArrayList<>();
+
+        Preconditions.checkNotNull(sourceDatastore);
```

Comment:
instead of Preconditions.checkNotNull use getDataStoreNode(requireNonNull(sourceDataStore), where requireNonNull is static import of Objects.requireNonNull

Reply:
Please, see the new commit.